salesagility/SuiteCRM
# Feature/email client - stripping out and starting again

## Description



This change is the first step to creating the new email client.

# Goals
* use the existing views in SuiteCRM
* keep the views consistent across SuiteCRM

# Changes
- I have stripped out the old email client.
- I have added some new features to the list view.

Using the listviewdefs.php you can now:
* add buttons to the list view (in the same way you do for the edit view)
* include javascript to handle the buttons (in the same way you do for the edit view)
* hide the edit view link (in addition to changing it in the sugar view)
* hide the header label (used for the indicator fields)


Example of the listviewdefs.php:
```
$viewdefs['Emails']['ListView'] = array(
    'templateMeta' => array(
        'form' => array(
            'buttons' =>
                array(
                    array(
                        'customCode' => '<a class="btn" data-action="compose-email">{$MOD.LBL_COMPOSEEMAIL} <span class="glyphicon glyphicon-envelope"></span></a>'
                    ),
                ),
        ),
        'includes' => array(
            0 =>
                array(
                    'file' => 'modules/Emails/javascript/list.view.js',
                ),
        ),
        'options' => array(
            'hide_edit_link' => true
        )
    )
);
```
```
'STATUS'=> array(
        'width' => '32',
        'label' => 'LBL_LIST_STATUS',
        'default' => false,
        'hide_header_label' => false
    ),
```
I have added two function fields to the list view for emails. An indicator for new and flagged emails and an indicator for when the email has attachments.

## Motivation and Context

* Feature for 7.9

## How To Test This

* Rebuild Sass
* Repair and Rebuild
* Set up scheduler / cron
* Create a group email
* Check that you can access the list view when you navigate to emails
* Check that email templates is now treated as it's own module.

![screenshot_20170303_094248](https://cloud.githubusercontent.com/assets/12231216/23546156/31f20002-fff6-11e6-8260-50d4e679c0e6.png)

**Please note that this is a work in progress. Some of the functionality is missing.**

## Types of changes


### Final checklist
